plugins/image-source/obs-slideshow.c

```diff
@@ -861,15 +957,16 @@ static obs_properties_t *ss_properties(void *data)
 
 	if (ss) {
 		pthread_mutex_lock(&ss->mutex);
-		if (ss->files.num) {
-			struct image_file_data *last = da_end(ss->files);
+		if (ss->paths.num) {
+			char *path_ = bstrdup(da_end(ss->paths));
 			const char *slash;
 
-			dstr_copy(&path, last->path);
+			dstr_copy(&path, path_);
```

Comment:
The bstrdup/bfree feels unnecessary here.  Should just do:

```C
dstr_copy(&path, da_end(ss->paths));
```

Reply:
When I try your code I get a segmentation fault.